service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ReviewCMOControllerAboutToStartTest.java

```diff
@@ -0,0 +1,103 @@
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(ReviewCMOController.class)
+@OverrideAutoConfiguration(enabled = true)
+public class ReviewCMOControllerAboutToStartTest extends AbstractControllerTest {
+
+    @MockBean
+    private DocumentSealingService documentSealingService;
+
+    protected ReviewCMOControllerAboutToStartTest() {
+        super("review-cmo");
+    }
+
+    @Test
+    void shouldReturnCorrectMapWhenMultipleCMOsReadyForApproval() {
+        DocumentReference order = TestDataHelper.testDocumentReference();
+        List<Element<CaseManagementOrder>> draftCMOs = List.of(
+            element(buildCMO("Test hearing 21st August 2020", order)),
+            element(buildCMO("Test hearing 9th April 2021", order)));
+
+        CaseData caseData = CaseData.builder().draftUploadedCMOs(draftCMOs).build();
+
+        AboutToStartOrSubmitCallbackResponse response = postAboutToStartEvent(asCaseDetails(caseData));
+
+        DynamicList dynamicList = DynamicList.builder()
+            .value(DynamicListElement.EMPTY)
+            .listItems(List.of(
+                DynamicListElement.builder()
+                    .code(draftCMOs.get(0).getId())
+                    .label("Test hearing 21st August 2020")
+                    .build(),
+                DynamicListElement.builder()
+                    .code(draftCMOs.get(1).getId())
+                    .label("Test hearing 9th April 2021")
+                    .build()
+            ))
+            .build();
+
+        CaseData responseData = mapper.convertValue(response.getData(), CaseData.class);
+
+        assertThat(responseData.getNumDraftCMOs()).isEqualTo("MULTI");
+        assertThat(responseData.getCmoToReviewList()).isEqualTo(
+            mapper.convertValue(dynamicList, new TypeReference<Map<String, Object>>() {}));
+    }
+
+    @Test
+    void shouldReturnCorrectMapWhenOneDraftCMOReadyForApproval() {
+        DocumentReference order = TestDataHelper.testDocumentReference();
+        List<Element<CaseManagementOrder>> draftCMOs = List.of(
+            element(buildCMO("Test hearing 21st August 2020", order)));
+
+        CaseData caseData = CaseData.builder().draftUploadedCMOs(draftCMOs).build();
+
+        AboutToStartOrSubmitCallbackResponse response = postAboutToStartEvent(asCaseDetails(caseData));
+
+        CaseData responseData = mapper.convertValue(response.getData(), CaseData.class);
+
+        assertThat(responseData.getNumDraftCMOs()).isEqualTo("SINGLE");
+        assertThat(responseData.getReviewCMODecision().getHearing()).isEqualTo("Test hearing 21st August 2020");
+        assertThat(responseData.getReviewCMODecision().getDocument()).isEqualTo(order);
+    }
+
+    @Test
+    void shouldReturnCorrectMapWhenNoDraftCMOsReadyForApproval() {
+        CaseData caseData = CaseData.builder().draftUploadedCMOs(List.of()).build();
+
+        AboutToStartOrSubmitCallbackResponse response = postAboutToStartEvent(asCaseDetails(caseData));
```

Comment:
could be, it is better to avoid reflection (.extracting("numDraftCMOs")):        
CaseData updateCaseData = extractCaseData(postAboutToStartEvent(asCaseDetails(caseData)));

        assertThat(updateCaseData.getNumDraftCMOs()).isEqualTo("NONE");